micro-rdk/src/common/config.rs

```diff
@@ -354,50 +590,66 @@ impl<'a> ConfigType<'a> {
             Self::Dynamic(cfg) => cfg.get_attribute::<T>(key),
         }
     }
+    #[deprecated(since = "0.5.1", note = "get_type() is deprecated use get_subtype()")]
     pub fn get_type(&self) -> &str {
         match self {
-            Self::Dynamic(cfg) => cfg.get_type(),
+            Self::Dynamic(cfg) => cfg.get_resource_name().get_subtype(),
+        }
+    }
+    pub fn get_subtype(&self) -> &str {
+        match self {
+            Self::Dynamic(cfg) => cfg.get_resource_name().get_subtype(),
         }
     }
 }
 
 pub trait Component {
+    fn get_resource_name(&self) -> &ResourceName;
+    fn get_model(&self) -> &Model;
+    #[deprecated(
```

Comment:
@acmorrow must be so proud

Reply:
Nothing makes me happier than a well executed deprecation!